core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -436,16 +436,14 @@ static TableMetadata fromJson(String metadataLocation, JsonNode node) {
     long currentSnapshotId = JsonUtil.getLong(CURRENT_SNAPSHOT_ID, node);
     long lastUpdatedMillis = JsonUtil.getLong(LAST_UPDATED_MILLIS, node);
 
-    Map<String, SnapshotRef> refs;
+    SnapshotOperations.Builder snapshotOperationsBuilder = SnapshotOperations.buildFromEmpty();
+
     if (node.has(REFS)) {
-      refs = refsFromJson(node.get(REFS));
+      snapshotOperationsBuilder.refs(refsFromJson(node.get(REFS)));
```

Comment:
Instead of passing a map back from `refsFromJson`, why not pass the `snapshotOperationsBuilder` to the parser method?